mmv1/products/netapp/volume.yaml

```diff
@@ -245,6 +264,32 @@ properties:
     description: |
       Reports the resource name of the Active Directory policy being used. Inherited from storage pool.
     output: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'restoreParameters'
+    description: |-
+      Used to create this volume from a snapshot (= cloning) or an backup.
+    immutable: true
+    # This parameter is only used at CREATE. READs will omit it.
+    ignore_read: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'sourceSnapshot'
+        description: |-
+          Full name of the snapshot to use for creating this volume.
+          `source_snapshot` and `source_backup` cannot be used simultaneously.
+          Format: `projects/{{project}}/locations/{{location}}/volumes/{{volume}}/snapshots/{{snapshot}}`.
+        conflicts:
+          - restore_parameters.0.source_backup
+        immutable: true
+      - !ruby/object:Api::Type::String
+        name: 'sourceBackup'
```

Comment:
Can we set this in a test?

Reply:
Not yet. We don't have a backup resource yet to create backups to restore from.
Backup is a bit complex. I have a draft for backup, which I didn't PR yet, since it needs more work. CRUD for backup isn't that complex itself, but the interaction between vault, backups and volumes is a bit complex. I want to make backupConfig parameter of volume work too first (making good progress, may need to update this PR to include it today or tomorrow). Also note that backup tests will extend test runtime for many minutes.

Back to this one: We currently cannot test it. I can remove it for now, but voted to keep it in, since the TF interaction is pretty simple. Specify an valid URN for a backup. There isn't too much which can go wrong in TF here, right?
What is your take: Omit it for now, or leave it in with the small risk that it might fail?

Reply:
Sorry for the late reply.
Can you test it locally, like create a backup resource not via Terraform and use that for `sourceBackup` for volume creation via Terraform. I think we can keep it and local testing should be sufficient.